lib/createTopLevelExpect.js

```diff
@@ -1429,7 +1430,7 @@ expectPrototype._executeExpect = function(
     }
   }
 
-  if (assertionRule.expect && assertionRule.expect !== this) {
+  if (assertionRule.expect && assertionRule.expect !== this._topLevelExpect) {
```

Comment:
I'm not entirely sure I understand why this one is a safe swap.

Reply:
I don't think it's exactly identical, but I'm fairly sure it's safe and also saves a recursive call when redelegating to an assertion registered in an `expect.child` from a nested `expect`.